atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/compact/BackgroundCompactor.java

```diff
@@ -88,8 +91,14 @@
     public void close() {
         log.info("Closing BackgroundCompactor");
         daemon.interrupt();
+        // Ensure we do not accidentally abort shutdown if any code incorrectly swallows InterruptedExceptions
+        // on the daemon thread.
+        shuttingDown.countDown();
         try {
             daemon.join();
+            if (daemon.isAlive()) {
```

Comment:
From join docs, it seems like this shouldn't occur. I do see that we have similar code in place for sweep though.